### ISSUE TITLE
Add FIM registry wildcard monitoring test support

### ISSUE THREAD

**Issue by Deblintrake09** [wazuh/wazuh-qa#3693]
| Target version | Related issue | Related PR/dev branch |
|--------------------|--------------------|-----------------|
|  4.5.0               |  https://github.com/wazuh/wazuh/issues/14321     |  https://github.com/wazuh/wazuh/pull/15852      |

## Description
This Issue aims to create IT tests for the new Wildcard support for FIM registry monitoring.

## Configurations
- Configure Registry path using wildcards in different configurations.

## Proposed test cases

- Base Test Case - Tier 1:
   - Scenario: One star in the path
```
Given I want to monitor everything under a root key
And I have entered ROOTKEY\*
When The configuration file has been read
Then All valid paths will be monitorable
Examples:
  | Path | Result |
  | HKEY_CURRENT_CONFIG\* | HKEY_CURRENT_CONFIG\Software,HKEY_CURRENT_CONFIG\System  |
```
- Tier 2:

   - Scenario: Two o more star in the path
```
Given I want to monitor everything under a root key and the subkeys
And I have entered ROOTKEY\*\* or ROOTKEY\*\Matcher
When The configuration file has been read
Then All valid paths will be monitorable
Examples:
  | Path | Result |
  | HKEY_CURRENT_CONFIG\*\* | HKEY_CURRENT_CONFIG\Software\Fonts,HKEY_CURRENT_CONFIG\System\CurrentControlSet  |
```

-   Scenario: A star in the beggining of the path
```
Given I want to monitor everything
And I have entered *\Software\ or *\*
When The configuration file has been read
Then An error will be raised
Examples:
  | Path | Result |
  | *\Software | Error: Undefined root key  |
```

   - Scenario: One question mark in the path
```
Given I want to monitor everything that matches with something
And I have entered ROOTKEY\SOME?\
When The configuration file has been read
Then All valid paths that matches with the question mark will be monitorable 
Examples:
  | Path | Result |
  | HKEY_CURRENT_CONFIG\Soft? | HKEY_CURRENT_CONFIG\Software  |
```

   - Scenario: Two o more question mark in the path
```
Given I want to monitor everything that matches with something
And I have entered ROOTKEY\SOME?\SOME?
When The configuration file has been read
Then All valid paths that matches with the question mark will be monitorable 
Examples:
  | Path | Result |
  | HKEY_CURRENT_CONFIG\Soft?\Goo? | NULL: No paths matches with the condition  |
```

   - Scenario: A question mark in the beggining of the path
```
Given I want to monitor everything
And I have entered HKEY?\SOMETHING\
When The configuration file has been read
Then An error will be raised
Examples:
  | Path | Result |
  | HKEY?\ | Error: Undefined root key  |
```
   - Scenario: A star and question mark in the path
```
Given I want to monitor everything
And I have entered ROOTKEY\SOMETHING?\* or ROOTKEY\*\SOMETHING?
When The configuration file has been read
Then All valid paths that matches with the question mark will be monitorable 
Examples:
  | Path | Result |
  | HKEY_CURRENT_CONFIG\Soft?\* | HKEY_CURRENT_CONFIG\Software\Fonts  |
  | HKEY_CURRENT_CONFIG\*\F? | HKEY_CURRENT_CONFIG\Software\Fonts  |

**Comment by damarisg** [comment]
I saw each proposed case, however, I think in this case, we can reduce these to the cases:

`Tier 1:`
- Case 1: HKEY_CURRENT_CONFIG\*
- Case 2: HKEY_CURRENT_CONFIG\Soft?

`Level 2:`
- ?HKEY_CURRENT_CONFIG\*\\*
-HKEY_CURRENT_CONFIG\Soft?\Goo?\*

It contains different combinations that check the same cases.

**Comment by Deblintrake09** [comment]
## Update 04/01/2022
- Research development
- Start designing tests modules

**Comment by Deblintrake09** [comment]
## Update 11/01/2022
- Try to generate packages :red_circle: 
- Asked Core about errors
- Continue designing test module

## Update 12/01/2022
- Continue working on tests
- Found Flaky behavior - Ask Core
- Refactor framework functions

**Comment by Deblintrake09** [comment]
## Update 11/01/2022
- Continue working on tests
- Flaky behavior continues
- Found Segfault in Wazuh Agent
- Blocking test until Core fixes segfaults

**Comment by Deblintrake09** [comment]
### Update 23/01/2023
- Generated packags
- Continue developing tests

**Comment by Deblintrake09** [comment]
### Update 26/01/2023
- Continue developing tests
 - Debug errors with `registry_value_update`

**Comment by Deblintrake09** [comment]
### Update 10/02/2023
- Ask core about windows event logs errors. Have not been fixed

**Comment by Deblintrake09** [comment]
### Update 17/02/2023
- Errors have been fixed. New ITs working as expected.
- Blocked as Jenkins is broken and cannot launch Instances.

**Comment by Deblintrake09** [comment]
## Update 09/06/2023
- Applied requested chagnes

**Comment by Deblintrake09** [comment]
Merge was reverted because Core development is not ready to merge.

**Comment by Deblintrake09** [comment]
## Update 04/07/2023
- Created new PR #4270 (recreated branch) and deleted old Merged PR #4220
